akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGraphCompileSpec.scala

```diff
@@ -227,10 +226,10 @@ class FlowGraphCompileSpec extends AkkaSpec {
 
     "build with implicits and variance" in {
       FlowGraph.closed() { implicit b ⇒
-        def appleSource = b.add(Source(PublisherProbe[Apple]))
-        def fruitSource = b.add(Source(PublisherProbe[Fruit]))
-        val outA = b add Sink(SubscriberProbe[Fruit]())
-        val outB = b add Sink(SubscriberProbe[Fruit]())
+        def appleSource = b.add(Source(TestPublisher.probe[Apple]))
```

Comment:
I experimented with extending a Sink and while not extending Sink itself (since it is final) I created a new Module for TestSink.probe which allows doing assertions in fluent fashion like so:
``` scala
FlowGraph.closed(in1, in2, TestSink.probe[String])((i1, i2, o) ⇒ o) { implicit b ⇒
  (in1, in2, o) ⇒
    val m1 = b.add(fairString)
    val m2 = b.add(fairString)

    // format: OFF
    in1.outlet ~> m1.in(0)
    in2.outlet ~> m1.in(1)

    Source(List("A", "B", "C", "D", "E", "F")) ~> m2.in(0)
                                        m1.out ~> m2.in(1)
                                                  m2.out ~> o.inlet
  // format: ON
}.run()
 .request(20)
 .expectNextUnordered("a", "b", "c", "d", "e", "f", "A", "B", "C", "D", "E", "F")
 .expectComplete()
```